src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CloseReferralController.java

```diff
@@ -0,0 +1,166 @@
+        @ApiResponse(responseCode = "200", description = "Accessed successfully",
+            content = {
+                @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = CCDCallbackResponse.class))
+            }),
+        @ApiResponse(responseCode = "400", description = "Bad Request"),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<CCDCallbackResponse> initHearingAndReferralDetails(
+        @RequestBody CCDRequest ccdRequest,
+        @RequestHeader(value = "Authorization") String userToken) {
+
+        if (!verifyTokenService.verifyTokenSignature(userToken)) {
+            log.error(INVALID_TOKEN, userToken);
+            return ResponseEntity.status(FORBIDDEN.value()).build();
+        }
+
+        CaseData caseData = ccdRequest.getCaseDetails().getCaseData();
+        caseData.setCloseReferralHearingDetails(referralHelper.populateHearingReferralDetails(caseData));
+        return getCallbackRespEntityNoErrors(caseData);
+    }
+
+    /**
+     * Called at the end of Close Referral event, it sets the referral status to Closed.
+     * @param ccdRequest holds the request and case data
+     * @param userToken  used for authorization
+     * @return Callback response entity with case data and errors attached.
+     */
+    @PostMapping(value = "/aboutToSubmit", consumes = APPLICATION_JSON_VALUE)
+    @Operation(summary = "")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Accessed successfully",
+            content = {
+                @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = CCDCallbackResponse.class))
+            }),
+        @ApiResponse(responseCode = "400", description = "Bad Request"),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<CCDCallbackResponse> aboutToSubmitReferralReply(
+        @RequestBody CCDRequest ccdRequest,
+        @RequestHeader(value = "Authorization") String userToken) {
+
+        if (!verifyTokenService.verifyTokenSignature(userToken)) {
+            log.error(INVALID_TOKEN, userToken);
+            return ResponseEntity.status(FORBIDDEN.value()).build();
+        }
+
+        CaseData caseData = ccdRequest.getCaseDetails().getCaseData();
+        referralHelper.setReferralStatusToClosed(caseData);
+        referralHelper.clearCloseReferralDataFromCaseData(caseData);
+        return getCallbackRespEntityNoErrors(caseData);
+    }
+
+    /**
+     * Called after submitting a close referral event.
+     * Returns the confirmation header and body
+     *
+     * @param ccdRequest holds the request and case data
+     * @param userToken  used for authorization
+     * @return Callback response entity with confirmation header and body
+     */
+    @PostMapping(value = "/completeCloseReferral", consumes = APPLICATION_JSON_VALUE)

```

Comment:
would a url of "submitted" be more aligned with other urls (like aboutToStart etc...)?